daemon/mgr/container.go

```diff
@@ -1215,118 +1217,197 @@ func (mgr *ContainerManager) updateContainerResources(c *Container, resources ty
 	return nil
 }
 
-// Upgrade upgrades a container with new image and args.
-func (mgr *ContainerManager) Upgrade(ctx context.Context, name string, config *types.ContainerUpgradeConfig) error {
-	c, err := mgr.container(name)
+func (mgr *ContainerManager) getImageRef(ctx context.Context, image string) (digest.Digest, string, error) {
```

Comment:
I do not think this is function of ContainerManager. It is better to locate it in ImageManager. So we should use `containerMgr.ImageMgr.GetImageRef`. WDYT? @HusterWan

Reply:
Good advice, i will put those function to `ImageMgr`